### ISSUE TITLE
v-intersection typescript is incorrect when using a function that takes 

### ISSUE THREAD

**Issue by hawkeye64** [quasarframework/quasar#16867]
### What happened?

```
Argument of type '(entry: IntersectionObserverEntry) => boolean' is not assignable to parameter of type 'IntersectionValue'.
  Type '(entry: IntersectionObserverEntry) => boolean' is not assignable to type '() => void'.ts(2345)

(property) onFormVisible: (entry: IntersectionObserverEntry) => boolean
```

The function looks like this:
```ts
function onFormVisible(entry: IntersectionObserverEntry): boolean {
  visible.value = entry.isIntersecting;
  return true;
}
```

It is used in the template like this:
```html
<div v-intersection="onFormVisible" class="full-width">
```

Screenshot:
![image](https://github.com/quasarframework/quasar/assets/10262924/e39e2aca-285e-465f-b050-c0943c900dd7)

### What did you expect to happen?

No typescript warnings

### Reproduction URL

https://no-example-available.com

### How to reproduce?

If you take the very first example at https://quasar.dev/vue-directives/intersection#basic and convert it to typescript, the issue will occur

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Directives (quasar), TypeScript Support

### Quasar info output

```Shell
$ quasar info

Operating System - Linux(6.5.0-10013-tuxedo) - linux/x64
NodeJs - 18.16.0

Global packages
  NPM - 9.5.1
  yarn - 1.22.19
  @quasar/cli - 2.3.0
  @quasar/icongenie - 3.1.1
  cordova - Not installed

Important local packages
  quasar - 2.14.3 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.7.3 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.16.9 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.4.15 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.2.5
  pinia - Not installed
  vuex - Not installed
  vite - 2.9.17 -- Native-ESM powered web dev build tool
  eslint - 8.56.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*
```

### Additional context

The `v-intersection` examples don't have any external examples to use (like CodePen)

**Comment by hawkeye64** [comment]
A little more info:
![image](https://github.com/quasarframework/quasar/assets/10262924/4fee17c4-43f3-422a-abcc-de959581ca69)

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.14.7
